sparse/core.py

```diff
@@ -681,6 +686,8 @@ def tensordot(a, b, axes=2):
 
 
 def dot(a, b):
+    if not hasattr(a, 'ndim') or not hasattr(b, 'ndim'):
+        return NotImplemented
```

Comment:
I think this should be ``raise NotImplementedError``, because ``dot`` is often called on its own. You could catch the ``NotImplementedError`` and ``return NotImplemented`` within ``__matmul__`` if you want to use it.